src/core/infer_request.cc

```diff
@@ -180,7 +248,6 @@ InferenceRequest::Release(
   // and the callback may interact with upper level trace.
   if (request->trace_ != nullptr) {
     request->trace_->ReportNow(TRITONSERVER_TRACE_REQUEST_END);
-    InferenceTrace::Release(std::move(request->trace_));
```

Comment:
You should "request->trace_ = nullptr" to drop the shared_ptr reference. That would be the equivalent of the current code.